src/main/java/org/broadinstitute/hellbender/tools/walkers/haplotypecaller/ReferenceConfidenceModel.java

```diff
@@ -0,0 +1,547 @@
+                // as this is the most conservative estimate of our certainty that we are hom-ref.
+                // For example, if the SNP PLs are 0,10,100 and the indel PLs are 0,100,1000
+                // we are very certain that there's no indel here, but the SNP confidence imply that we are
+                // far less confident that the ref base is actually the only thing here.  So we take 0,10,100
+                // as our GLs for the site.
+                final GenotypeLikelihoods leastConfidenceGLs = getGLwithWorstGQ(indelGLs, snpGLs);
+
+                gb.GQ((int) (-10 * GenotypeLikelihoods.getGQLog10FromLikelihoods(GenotypeType.HOM_REF.ordinal()-1, leastConfidenceGLs.getAsVector())));
+                gb.PL(leastConfidenceGLs.getAsPLs());
+
+                vcb.genotypes(gb.make());
+                results.add(vcb.make());
+            }
+        }
+
+        return results;
+    }
+
+    /**
+     * Get the GenotypeLikelihoods with the least strong corresponding GQ value
+     * @param gl1 first to consider (cannot be null)
+     * @param gl2 second to consider (cannot be null)
+     * @return gl1 or gl2, whichever has the worst GQ
+     */
+    @VisibleForTesting
+    GenotypeLikelihoods getGLwithWorstGQ(final GenotypeLikelihoods gl1, final GenotypeLikelihoods gl2) {
+        return GenotypeLikelihoods.getGQLog10FromLikelihoods(GenotypeType.HOM_REF.ordinal()-1, gl1.getAsVector())
+                >
+               GenotypeLikelihoods.getGQLog10FromLikelihoods(GenotypeType.HOM_REF.ordinal()-1, gl2.getAsVector()) ? gl1 : gl2;
+    }
+
+    /**
+     * Get indel PLs corresponding to seeing N nIndelInformativeReads at this site
+     *
+     * @param nInformativeReads the number of reads that inform us about being ref without an indel at this site
+     * @param ploidy the requested ploidy.
+     * @return non-null GenotypeLikelihoods given N
+     */
+    @VisibleForTesting
+    GenotypeLikelihoods getIndelPLs(final int ploidy, final int nInformativeReads) {
+        return indelPLCache(ploidy, nInformativeReads > MAX_N_INDEL_INFORMATIVE_READS ? MAX_N_INDEL_INFORMATIVE_READS : nInformativeReads);
+    }
+
+    private GenotypeLikelihoods indelPLCache(final int ploidy, final int nInformativeReads) {
+        return initializeIndelPLCache(ploidy)[nInformativeReads];
+    }
+
+    private GenotypeLikelihoods[] initializeIndelPLCache(final int ploidy) {
+
+        if (indelPLCache.length <= ploidy) {
+            indelPLCache = Arrays.copyOf(indelPLCache, ploidy << 1);
+        }
+
+        if (indelPLCache[ploidy] != null) {
+            return indelPLCache[ploidy];
+        }
+
+        final double denominator =  - MathUtils.log10(ploidy);
+        final GenotypeLikelihoods[] result = new GenotypeLikelihoods[MAX_N_INDEL_INFORMATIVE_READS + 1];
+        result[0] = GenotypeLikelihoods.fromLog10Likelihoods(new double[ploidy + 1]);
```

Comment:
Add a one line comment stating that an array full of zeros is the right answer for result[0].